api/src/test/java/org/openmrs/api/UserServiceTest.java

```diff
@@ -280,7 +282,43 @@ public void createUser_shouldNotAllowSystemIdEqualsUsernameWithLuhnCheckDigit()
 						newUser.getSystemId()));
 		userService.createUser(newUser, SOME_VALID_PASSWORD);
 	}
-
+	@Test
+	public void createUser_shouldNotAllowCreatingUserWithPrivilegeCurrentUserDoesNotHave() throws IllegalAccessException {
+		//setup the currently logged in user
+				User currentUser = new User();
+				Role userRole = new Role("User Adder");
+				userRole.setRole(RoleConstants.AUTHENTICATED);
+				userRole.addPrivilege(new Privilege("Add Users"));
+				currentUser.addRole(userRole);
+				// setup our expected exception
+				// we expect this to fail because the currently logged-in user lacks a privilege to be
+				// assigned to the new user
+				expectedException.expect(APIException.class);
+				expectedException.expectMessage("You must have privilege {0} in order to assign it");
+				// set current user to the user defined above
+				withCurrentUserAs(currentUser, () -> {
+					// create a role to assign to the new user
+					Role role = new Role();
+					role.setRole(RoleConstants.AUTHENTICATED);
+					role.addPrivilege(new Privilege("Custom Privilege"));// add a privilege to the role
+
+					// create our new user object with the required fields
+					User u = new User();
+					u.setPerson(new Person());
+					// assign the specified role to the user
+					u.addName(new PersonName("Benjamin", "A", "Wolfe"));
+					u.setUsername("bwolfe");
+					u.getPerson().setGender("M");
+					u.addRole(role);
+					// here we expect the exception to be thrown
+					userService.createUser(u, "Openmr5xy");
+				});
+	}
+				
+	@Test
+	public void createUser_shouldNotAllowAssigningSuperUserRoleIfCurrentUserDoesnotHaveAssignSystemDeveloperPrivileges() throws IllegalAccessException {
+				
```

Comment:
The body here should look identical to the above test except that we should assign two privileges to `role` and we should expect the `User you must have the following privileges in order to assign them: {0}` message.

Reply:
Am trying to do as you suggested but there is problem with defining method

Reply:
What kind of problem?

Reply:
Everything is okey now, i hope to finish tests on sunday

Reply:
cc.@ibacher, kindly look through this code, thanks